### DISCUSSION TITLE
Parser thread safe?

### DISCUSSION THREAD

**Discussion by SebastianStehle** [sebastienros/jint#1412] [Q&A]
Hi,
I upgraded from `beta-2036` to `beta-2046` and because the parser takes the options as ctor arguments, I thought, it is thread safe now. Is this the case?

If not, would your recommend to use an object pool?

**Comment by lahma** [comment]
If we are talking about `JavaScriptParser`, it's not thread safe. The options should be thread safe as it's immutable record by nature. Parser should be cheap to instantiate and run so I'd just keep the parsed AST around if your scripts don't change.
